src/app/plugins-initializer.ts

```diff
@@ -0,0 +1,69 @@
+import { APP_INITIALIZER, ValueProvider } from '@angular/core';
+import { getAngularJSGlobal } from '@angular/upgrade/static';
+import { AJS_MODULE } from './modules/base/shared/constants';
+
+export interface PluginDefinition {
+  angularModules: string[];
+  scripts: string[];
+}
+
+const loadScript = (fileName: string): Promise<unknown> =>
+  new Promise<unknown>((resolve) => {
+    const callback = () => {
+      console.log(`Loaded plugin script ${fileName}`);
+      resolve(true);
+    };
+    const script = document.createElement('script');
+    script.setAttribute('type', 'text/javascript');
+    script.setAttribute('src', fileName);
+    //@ts-ignore
+    script.onreadystatechange = callback;
+    script.onload = callback;
+    document.querySelector('head')!.appendChild(script);
+  });
+
+const loadSinglePlugin = async (pluginDefinition: PluginDefinition): Promise<string[]> => {
+  try {
+    const scriptsLoad = pluginDefinition.scripts.map((script) => loadScript(script));
+    await Promise.all(scriptsLoad);
+  } catch (e) {
+    console.log(`Module(s) ${pluginDefinition.angularModules.join(', ')} load fail`);
+    console.log(e);
```

Comment:
I would suggest to use the second parameter of console.log when possible:

`console.log(`Module(s) ${pluginDefinition.angularModules.join(', ')} load fail`, e);`